src/api/ucc.h

```diff
@@ -0,0 +1,709 @@
+} ucc_thread_mode_t;
+
+typedef enum {
+	UCC_NO_SYNC_COLLECTIVES = 0,
+	UCC_SYNC_COLLECTIVES = 1
+} ucc_coll_sync_type_t;
+
+/**
+ * @brief UCC library initializatoin parameters
+ */
+
+typedef struct ucc_lib_params {
+    uint64_t                mask;
+    ucc_thread_mode_t       thread_mode;
+    ucc_coll_type_t         coll_types;
+    ucc_reduction_op_t      reduction_types;
+    ucc_coll_sync_type_t    sync_type;
+} ucc_lib_params_t;
+
+
+typedef struct ucc_lib_attribs {
+    uint64_t                mask;
+    ucc_thread_mode_t       thread_mode;
+    ucc_coll_type_t         coll_types;
+    ucc_reduction_op_t      reduction_types;
+    ucc_coll_sync_type_t    sync_type;
+} ucc_lib_attribs_t;
+
+
+/**
+ * @ingroup UCC_LIB
+ * @brief @ref ucc_config_read allocates the @ref ucc_lib_config_t structure and
+ * fetches the configuration values from the run-time environment. The run-time
+ * environment supported are environment variables or a configuration file.
+ *
+ *   @param [out] config_p      Pointer to configuration descriptor as defined by
+ *                              @ref ucc_lib_config_t.
+ *   @param [out] env_prefix    If not NULL, the routine searches for the
+ *                              environment variables with the prefix UCC_<env_prefix>. Otherwise, the
+ *                              routines search for the environment variables that start with the prefix @
+ *                              UCC_.
+ *   @param [in]  filename      If not NULL, read configuration values from the
+ *                              file defined by @e filename. If the file does not exist, it will be ignored
+ *                              and no error will be reported to the user.
+ *
+ * @return Error code as defined by @ref ucc_status_t
+ */
+
+ucc_status_t ucc_lib_config_read(const char *env_prefix, const char *filename,
+                                 ucc_lib_config_t **config_p);
+
+/**
+ * @ingroup UCC_LIB
+ * @brief Release configuration descriptor
+ *
+ * The routine releases the configuration descriptor that was allocated through
+ * @ref ucc_lib_config_read "ucc_lib_config_read()" routine.
+ *
+ * @param [in] config        Pointer to the configuration descriptor to be released. 
+ *                           Configuration descriptor as defined by
+ */
+
+void ucc_lib_config_release(ucc_lib_config_t *config);
+
+/**
+ * @ingroup UCC_LIB
+ * @brief Print configuration information
+ *
+ * The routine prints the configuration information that is stored in
+ * @ref ucc_lib_config_h "configuration" descriptor.
+ *
+ * @param [in]  config        @ref ucc_lib_config_h "Configuration descriptor"
+ *                            to print.
+ * @param [in]  stream        Output stream to print the configuration to.
+ * @param [in]  title         Configuration title to print.
+ * @param [in]  print_flags   Flags that control various printing options.
+ */
+
+void ucc_lib_config_print(const ucc_lib_config_t *config, FILE *stream,
```

Comment:
ucc_confif_print_flags_t - is -it defined?

Reply:
It is defined in UCS. However, we will adapt the definition to suit our needs. I will add it to ucc_def.